### PR TITLE
Make UploadArea database id an integer sequence

### PR DESCRIPTION
**Pull request by sampierson**
rename the old 'id' column to 'uuid'

### CODE REVIEW

File: tests/unit/common/test_upload_area.py [hunk 1]
```diff
@@ -144,89 +143,89 @@ def test_unlock__with_locked_area__unlocks_area(self):
...
         }, fileinfo)
-        obj = self.upload_bucket.Object(f"{area_id}/some.json")
+        obj = self.upload_bucket.Object(f"{db_area.uuid}/some.json")
         self.assertEqual("exquisite corpse".encode('utf8'), obj.get()['Body'].read())
 
         file_id = f"{area.uuid}/{filename}"
         db_file = self.db.query(DbFile).get(file_id)
         self.assertEqual(16, db_file.size)
-        self.assertEqual(area_id, db_file.upload_area_id)
+        self.assertEqual(db_area.id, db_file.upload_area_id)
```
Comment (MDunitz) [inline tests/unit/common/test_upload_area.py]: should this be `db_area.uuid`?

**Comment by sampierson** [comment]
This got autoclosed because I deleted the base branch.  Reopening.

File: upload/common/upload_area.py [hunk 2]
```diff
@@ -267,20 +276,20 @@ def _retrieve_upload_area_deletion_lambda_timeout(self):
...
         }
 
-    def _db_record(self):
-        return self.db.get_pg_record('upload_area', self.uuid)
-
     def _create_record(self):
-        prop_vals_dict = self._format_prop_vals_dict()
-        self.db.create_pg_record("upload_area", prop_vals_dict)
+        prop_vals_dict = self._serialize()
+        del prop_vals_dict['id']
```
[the remark this answers is not in the document]
Comment (sampierson) [reply upload/common/upload_area.py]: I like it.  It will be so.

**Comment by sampierson** [comment]
Coping a convo with Madison from Slack to keep for posterity:

[Madison]
I was wondering what the purpose/reason for the new id column is. Possibly bc I’ve just been looking at the pr with lots of references to upload_area_id/upload_area_uuid /upload_area.db_id/area_uuid but I’m having some trouble understanding the distinction/need for the two columns

also unrelated, I googled this a little but wasnt able to find a definitive answer. At what point would we it be worth it to create a new index concurrently? 10x our current db size? 100x?

Sam Pierson [7:50 AM]
I’m starting to move the Upload DB to using Postgres integer SEQUENCEs for all primary keys.  This is standard practice in other ORMs I have used in the past (Rails ActiveRecord).  `Upload_area` is just the first one to get converted and was the one I used to prototype this in Python/SQLAlchemy.  The `File` table will be following shortly.  The impetus for doing this is that if you use some significant piece of your data model as the primary key, there is a chance that it may become non-unique in each row of your table.  This has now happened in the `file` table.   The current data item being used as the primary key in the `file` table is the S3 key of the file in the S3 bucket.  In the original database design this was unique to each row, but that is about to be no longer the case.  Rows in `file` will now only be unique based on the combination of `s3_key` and `s3_etag`.   It saves a ton of work and juggling later if you just us a primary key up front that is not tied to the data and guaranteed to be unique.  In this case it would have saved weeks of work.

I don’t have a good answer to your second question.  I think it is always a good practice to create indices concurrently, but in this case I tried it and it gave me an error (I forget what), and I don’t think it’s so critical yet that I think it is worth spending the time figuring it out. (edited)

**Comment by sampierson** [comment]
Note that tests for this branch will keep failing until I migrate the test database.  We currently don't do that as a part of tests.  Really, we should wipe the test database and migrate from scratch at the start of each test, but that's work for a different PR.  Meanwhile I just need to coordinate with everyone else.

**Comment by sampierson** [comment]
Re: codeclimate issue, see https://app.zenhub.com/workspaces/dcp-backlogs-5ac7bcf9465cb172b77760d9/issues/humancellatlas/upload-service/262

**Pull request merged**